server/consumer.go

```diff
@@ -1698,6 +1737,7 @@ func (o *consumer) info() *ConsumerInfo {
 
 	// If we are a pull mode consumer, report on number of waiting requests.
 	if o.isPullMode() {
+		o.expireWaiting()
```

Comment:
Does this mean that doing `sub.ConsumerInfo()` will expire the pending requests?

Reply:
Yes it will reclaim old requests here, and when we move to a wait state in the loop and gather, and we check them when we pop to deliver a message.

Reply:
I think this is good approach, it was needed some way to trigger the expiration of accumulated pending requests that have just timed out without a client issuing a 'get next' request. Now when the state is confirmed by using `nats consumer info` for example, the expired requests would no longer be accounted in the `NumWaiting` field.

Reply:
We used to expire pending requests sporadically and not on a regular time interval. I did not opt for a time interval per se but added a bunch of spots to where we do collect now.

1. Namely when we pop a request we now check inline, so no delivering to a request that we invalidate.
2. When the consumer's loopAndGather goes into a waiting state we check the pending list then as well.
3. When someone asks for consumer info on a pull consumer.
4.